tools/run_tests/xds_k8s_test_driver/framework/infrastructure/k8s.py

```diff
@@ -502,13 +590,68 @@ def delete_gamma_route(
         # TODO(sergiitk): [GAMMA] Can we call delete on dynamic_res.ResourceList
         #  to avoid no-member issues due to dynamic_res.Resource proxying calls?
         self._execute(
-            self.api_grpc_route.delete,  # pylint: disable=no-member
+            self.api_http_route.delete,  # pylint: disable=no-member
+            name=name,
+            namespace=self.name,
+            propagation_policy="Foreground",
+            grace_period_seconds=grace_period_seconds,
+        )
+
+    def delete_session_affinity_policy(
+        self,
+        name: str,
+        grace_period_seconds=DELETE_GRACE_PERIOD_SEC,
+    ) -> None:
+        self._execute(
+            self.api_session_affinity_policy.delete,
+            name=name,
+            namespace=self.name,
+            propagation_policy="Foreground",
+            grace_period_seconds=grace_period_seconds,
+        )
+
+    def delete_session_affinity_filter(
+        self,
+        name: str,
+        grace_period_seconds=DELETE_GRACE_PERIOD_SEC,
+    ) -> None:
+        self._execute(
+            self.api_session_affinity_filter.delete,
+            name=name,
+            namespace=self.name,
+            propagation_policy="Foreground",
+            grace_period_seconds=grace_period_seconds,
+        )
+
+    def delete_backend_policy(
+        self,
+        name: str,
+        grace_period_seconds=DELETE_GRACE_PERIOD_SEC,
+    ) -> None:
+        self._execute(
+            self.api_backend_policy.delete,
             name=name,
             namespace=self.name,
             propagation_policy="Foreground",
             grace_period_seconds=grace_period_seconds,
         )
 
+    def delete_pod_async(
```

Comment:
Is this intended to check-in/merge to master or just temporarily added for debugging?  I don't see this get called in other place.   Do I miss anything?

Reply:
This will be used for session draining tests, which I have temporarily deleted since they are not yet ready. See [this commit](https://github.com/grpc/grpc/pull/34387/commits/1712a0b3916394d5229ae2e39b0ff8e261aaba2c) for the usage.